### ISSUE TITLE
Implement programmatic change detection profiler

### ISSUE THREAD

**Issue by yjbanov** [angular/angular#3877]
The profiler would be accessible via dev tools. When a developer installs the profiler DI module they get access to global functions they could invoke directly from the dev tools console.

The profiler would be exposed under global namespace `ng` (which in the future could host more tools). To profile change detection specifically, a developer would enter:

```javascript
ng.profileChangeDetection();
```

The profiler would invoke `LifeCycle.tick()` at least 5 times and for at least 500ms. It will then compute the average and the standard deviation of amount of time it took to execute a single `tick`.

`profileChangeDetection` takes an optional named boolean parameter `record` that's `false` by default. When set to `true`, it records the profile as a CPU profile that could be inspected later in the "Profiles" tab of the dev tools:

```javascript
ng.profileChangeDetection({ record: true });
```

In the future, we may add more options as named parameters, such as custom number of `tick` runs or custom length of profile.

/cc @vsavkin, @mhevery

**Comment by PascalPrecht** [comment]
I love this. Same APIs could be used to make a decent browser extension out of it (or extending Batarang)

**Comment by mhevery** [comment]
@yjbanov Sounds good to me. Please use `ng` as the global object as it will be consistent with `ng.probe` etc...

**Comment by gkalpak** [comment]
@yjbanov, shouldn't this have been closed by 8dd6c46 ?

**Comment by yjbanov** [comment]
@gkalpak yep
